src/features/channels/components/channel-list/term-conditions/termConditions.tsx

```diff
@@ -0,0 +1,59 @@
+import { Button } from '@mui/material';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import Typography from '@mui/material/Typography';
+import * as React from 'react';
+import useStyles from './termConditions.styles';
+
+type Props = {
+  title: string;
+  message: string;
+  onClick?: () => void;
+};
+
+const TermConditions = ({ title, message, onClick }: Props) => {
+  const [open, setOpen] = React.useState(false);
+  const styles = useStyles();
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <Typography
+        variant="caption"
+        display="block"
+        className={styles.term}
+        onClick={handleClickOpen}
+      >
+        Terms and Conditions
+      </Typography>
+      <Dialog
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="alert-dialog-title"
+        aria-describedby="alert-dialog-description"
+      >
+        <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">{message}</DialogContentText>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleClose}>Disagree</Button>
+          <Button onClick={handleClose} autoFocus>
+            Agree
+          </Button>
+        </DialogActions>
```

Comment:
Maybe just put some dummy text in this modal and ditch the 'agree' and 'disagree' buttons because you're already inside the app, you cannot disagree once you're logged in

Reply:
Maybe "disagree and log out"? and "agree and stay"

Reply:
It just doesn't feel natural, I have never used an app that gave me the possibility to disagree with the terms and conditions when I'm already logged in. But yeah, we can talk about it in the next meeting